pkg/credstores/credstore.go

```diff
@@ -82,6 +94,45 @@ func Init(ctx context.Context, toolRegistries []string, dsn string, opts Options
 	}
 }
 
+func setUpAzureKeyVault(ctx context.Context, keyvaultName, keyName, keyVersion, configFile string) error {
+	if keyvaultName == "" || keyName == "" || keyVersion == "" {
+		return fmt.Errorf("missing Azure Key Vault configuration")
+	}
+
+	if err := os.Setenv("GPTSCRIPT_ENCRYPTION_CONFIG_FILE", configFile); err != nil {
+		return fmt.Errorf("failed to set GPTSCRIPT_ENCRYPTION_CONFIG_FILE: %w", err)
+	}
+
+	if err := os.WriteFile("/tmp/azure.json", []byte(`{"useManagedIdentityExtension": true}`), 0600); err != nil {
```

Comment:
Does it make sense to use `os.CreateTemp` instead of create a named temporary file?

Reply:
I think it's fine as it is. This is really only meant to be run in the container inside an Azure VM, so there's not gonna be a file already there.